devito/types/sparse.py

```diff
@@ -41,7 +41,8 @@ def __init_finalize__(self, *args, **kwargs):
         self._space_order = kwargs.get('space_order', 0)
 
         # Dynamically add derivative short-cuts
-        self._fd = generate_fd_shortcuts(self)
+        self._fd = generate_fd_shortcuts(self.dimensions, self.space_order,
+                                         to=kwargs.get('time_order', 0))
```

Comment:
that should only appear in a SparseTimeFunction

Reply:
Then would generate shortcuts twice.

Reply:
same as before, this should be moved within a `__fd_setup__` method

Reply:
fixed